static_frame/core/index_hierarchy.py

```diff
@@ -949,59 +1275,77 @@ def _ufunc_set(self,
             # if other is not an IndexHierarchy, do not try to propagate types
             index_constructors = None
 
-        return cls.from_labels(labels,
+        return cls._from_arrays(
+                labels,
+                name=self.name,
                 index_constructors=index_constructors,
-                depth_reference=self.depth)
+                depth_reference=self.depth,
+                )
 
+    # --------------------------------------------------------------------------
 
-    #---------------------------------------------------------------------------
-    def _drop_iloc(self, key: GetItemKeyType) -> 'IndexHierarchy':
-        '''Create a new index after removing the values specified by the loc key.
+    @property
+    def _index_constructors(self: IH) -> tp.Iterator[tp.Type[Index]]:
+        '''
+        Yields the index constructors for each depth.
+        '''
+        yield from (index.__class__ for index in self._indices)
+
+    def _drop_iloc(self: IH,
+            key: GetItemKeyType,
+            ) -> IH:
+        '''
+        Create a new index after removing the values specified by the iloc key.
         '''
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
```

Comment:
Why not cover this scenario with a test?

Reply:
I got tired of adding entire tests just to add coverage to these.

If I wasn't so concerned about performance, I would have designed it like this:

```python
def _update_array_cache(self):
    if not self._recache:
        return
    ...

def meth1(self):
    self._update_array_cache()
    ...

def meth2(self):
    self._update_array_cache()
    ...
```
However, since this is _slightly_ slower than the other approach (perform the boolean check _before_ the function call), it was not implemented this way.

An annoying side effect is that now it requires a how lot of redundant tests just to capture the branch, which after doing for a while, I got tired of and so I just went with `pragma: no-cover`

Reply:
I agree that using the in-function branch is best. But I do not think that these cases justify `pragma: no-cover`; the cases we use `pragma: no cover` are very few. You might find that you have put this line in functions that do not need it, or other situations that arise when you have deferred operations in this particular context.

Reply:
:cry: